### PR TITLE
OHRI-2077 Add unit tests for encounter tile in commons lib

### PR DESCRIPTION
**Pull request by CynthiaKamau**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes the ticket number in the format `OHRI-123 My PR title`.
- [ ] My work includes tests or is validated by existing tests.

## Summary

This is an initial pr for : Add unit tests for encounter tile in commons lib, i refactored the component and moved the data fetching section to the useLastEncounterHook

### CODE REVIEW

File: packages/esm-commons-lib/src/hooks/useLastEncounter.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  const query = `encounterType=${encounterType}&patient=${patientUuid}&limit=1&order=desc&startIndex=0`;
+    const endpointUrl = `/ws/rest/v1/encounter?${query}&v=${encounterRepresentation}`;
+
+  const { data, error, isValidating } = useSWR<{ data: { results: Array<OpenmrsEncounter> } }, Error>(
+    endpointUrl,
+    openmrsFetch,
+  );
+
+  return {
+    lastEncounter: data ? data?.data?.results.shift() : null,
```
Comment (eudson) [inline packages/esm-commons-lib/src/hooks/useLastEncounter.tsx:16]: @CynthiaKamau are you sure the first element of the array returned is the latest encounter? Can I assume that you taken care of that on the query?

File: packages/esm-commons-lib/src/components/encounter-tile/encounter-tile.component.tsx [hunk 2]
```diff
@@ -20,65 +20,61 @@ export interface EncounterTileProps {
...
+  patientUuid: string;
+  column: any;
+}
+
 export const EncounterTile: React.FC<EncounterTileProps> = ({ patientUuid, columns, headerTitle }) => {
-  const [lastEncounter, setLastEncounter] = useState(null);
-  const [isLoading, setIsLoading] = useState(true);
+  return (
+    <>
+      <Tile className={styles.tile}>
```
[the remark this answers is not in the document]
Comment (pirupius) [reply packages/esm-commons-lib/src/components/encounter-tile/encounter-tile.component.tsx:31]: I agree with this. Let's create a `commons.scss` file or maybe group similar styles into specific files. For example tiles can have a single file `tiles.scss`

File: packages/esm-commons-lib/src/components/encounter-tile/encounter-tile.test.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { act, render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
```
Comment (pirupius) [inline packages/esm-commons-lib/src/components/encounter-tile/encounter-tile.test.tsx:3]: we don't need this

File: packages/esm-commons-lib/src/components/encounter-tile/encounter-tile.component.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (CynthiaKamau) [reply packages/esm-commons-lib/src/components/encounter-tile/encounter-tile.component.tsx:30]: added a styleguide directory with the tile styles

File: packages/esm-commons-lib/src/components/encounter-tile/encounter-tile.test.tsx (hunk 3 above)
Comment (CynthiaKamau) [reply packages/esm-commons-lib/src/components/encounter-tile/encounter-tile.test.tsx:3]: if i remove it i get undefined for some jest functions

File: packages/esm-commons-lib/src/index.ts [hunk 4]
```diff
@@ -46,4 +46,5 @@ export * from './components/card-summary/summary-card.component';
 export * from './components/expandable-list-widget/expandable-list.component';
 export * from './components/extension-conditional-renderer/patient-based-extension-renderer';
 export * from './components/patient-table/patient-table.component';
-export * from './hooks/useLastEncounter';
\ No newline at end of file
+export * from './hooks/useLastEncounter';
+export * from './styleguide/tiles.scss';
```
Comment (pirupius) [inline packages/esm-commons-lib/src/index.ts:50]: We don't need this. These exports are for components that can be imported outside this esm for reuse.

Comment (CynthiaKamau) [reply packages/esm-commons-lib/src/index.ts]: done

**Review by pirupius — APPROVED** [review]

**Pull request merged**